src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,306 @@
+/**
+ * @file cmaes.hpp
+ * @author Kartik Nighania (GSoC 17 mentor Marcus Edel)
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+#include "random.hpp"
+
+namespace mlpack {
+namespace optimization {
+
+template<typename funcType>
+class CMAES
+{
+ public:
+CMAES(funcType& function, arma::mat& start, arma::mat& stdDivs,
+double iters = -1.0, double evalDiff = 1e-14);
+
+double Optimize(arma::mat& arr);
+
+size_t getDimension(void){ return N;}
+
+void getXBestEver(double *arr)
+{
+for (int i = 0; i < N; ++i) arr[i] = xBestEver[i];
+}
+
+void stopMaxFuncEvaluations(double evaluations)
```

Comment:
Can you adjust the function style here, e.g. use:

```
//! Get the ... parameter.
double MaxEvaluations() const { return stopMaxFunEvals; }
//! Modify the ... parameter.
double& MaxEvaluations() { return stopMaxFunEvals; }
```

to be consistent with the rest of the codebase? That applies for most of the following functions too.